### ISSUE TITLE
Sensors become stale and not unavailable and can't be deleted

### ISSUE THREAD

**Issue by golles** [home-assistant/iOS#3120]
**Device model, version and app version**

Model Name: Apple M3 Pro
macOS Version: 15.0.1
App Version: 2024.9.2
Safari Version: I don't use safari

**Home Assistant Core Version**

2024.10.4

**Describe the bug**
Sensors become stale and not unavailable and can't be deleted

**To Reproduce**
I had issues with my UPS and have connected it to my MacBook, I got a new one and also connected that to the Macbook to check it before installing. After disconnecting, the sensors remain active and can't be deleted.

**Expected behavior**
Something should happen with these sensors, I think what other integrations do is make them unavailable, which probably makes the most sense. 
In my case, I would see them deleted, if they would become unavailable I could delete them manually.

**Screenshots**

In HA the sensors remain there and look active:
<img width="307" alt="Scherm­afbeelding 2024-11-01 om 11 41 16" src="https://github.com/user-attachments/assets/64aafe35-6dfc-4696-8b82-cfe8fcf2a23f">

After restarting the app on my Macbook, the sensors were removed from here:
<img width="311" alt="Scherm­afbeelding 2024-11-01 om 11 41 01" src="https://github.com/user-attachments/assets/f678e35b-57ac-467a-a285-13d531c4bdaf">

The delete button isn't clickable, so can't delete them
<img width="567" alt="Scherm­afbeelding 2024-11-01 om 11 46 49" src="https://github.com/user-attachments/assets/ec161c52-854a-42ab-89c4-08dce611d7fe">

**Additional context**

**Comment by bgoncal** [comment]
Correct me if I'm wrong but your UPS entities have no relation to macOS App right? How did you add them to Home Assistant?

**Comment by golles** [comment]
I guess the app detects all batteries connected to the Mac? The only thing I did is plug in the USB cable.

**Comment by bgoncal** [comment]
Oh, sweet! I honestly didn't know this was a feature haha, I'll update you as soon as I have more info

**Comment by golles** [comment]
It is actually nice indeed, this is what I see in my Mac battery settings when the UPS is connected
<img width="862" alt="Scherm­afbeelding 2024-11-04 om 19 40 07" src="https://github.com/user-attachments/assets/ae547b1f-cc1d-4d8c-8121-3003f9b1a7dc">

**Comment by golles** [comment]
After I restarted the Mac, they still show up in HA.
<img width="312" alt="Scherm­afbeelding 2024-11-05 om 23 19 29" src="https://github.com/user-attachments/assets/3b405cc2-9fa9-414f-933c-64a09d37885f">
and they still look active, which still doesn't allow me to remove them.

I understand I can fix this by removing the integration for this Macbook, but that would be ignoring this bug.

But now I think of it, is this a bug from the app or within core?
As the app doesn't show these sensors anymore, shouldn't the integration in core detect them as stale and mark them as unavailable? What do you think?

**Comment by bgoncal** [comment]
Sorry, I didn't mean to close this bug with the workaround, just giving you short term solution.

I think iOS app needs to ask core to remove those sensors, but I am not sure, @dshokouhi what do you do in Android in such cases?

**Comment by dshokouhi** [comment]
> what do you do in Android in such cases?

ours are also not removed in these cases, @jpelgrom and I discussed this recently actually as he worked on the sync status between the server for when sensors get enabled/disabled

**Comment by jpelgrom** [comment]
> ours are also not removed in these cases

I don't believe there are sensors like these which can completely disappear on Android so not completely comparable. But yes, it disables/enables like other integrations do.

**Comment by qguv** [comment]
This problem still persists on macOS. Any sensor that was once enabled (i.e. the defaults), but has since been disabled in the companion app settings, shows up as "unavailable". Some examples:

<img width="916" alt="several disabled sensors marked as unavailable in the entities list" src="https://github.com/user-attachments/assets/edc8402f-1b74-4440-a3bc-11c68f966f20" />

If you try to delete them, you get the following message:

<img width="565" alt="You can only delete 0 of the 1 entities. The others require the integration to stop providing them, and sometimes a Home Assistant restart is needed. Are you sure you want to delete the deletable entities? Remove them from your dashboard and automations if they include these entities." src="https://github.com/user-attachments/assets/349f3ec7-0c55-4b0e-8e1f-d28f835f8a2c" />
